### ISSUE TITLE
[BFS] illegal memory access, top of pre-release branch

### ISSUE THREAD

**Issue by sgpyc** [gunrock/gunrock#200]
```
./bin/test_bfs_8.0_x86_64 market /data/gunrock_dataset/large/arabic-2005/arabic-2005.mtx --src=randomize2 --queue-sizing=6.5 --in-sizing=4 --iteration-num=16 --idempotence --64bit-SizeT --device=0,1 --do_a=100 --do_b=100 --partition-seed=1475611716 --src-seed=1475611723
Loading Matrix-market coordinate-formatted graph ...
  Reading directly from stored binary CSR arrays ...
Done reading (1s).

Degree Histogram (22744080 vertices, 631153669 edges):
    Degree   0: 3425212 (15.06%)
    Degree 2^0: 1733525 (7.62%)
    Degree 2^1: 1253448 (5.51%)
    Degree 2^2: 2020393 (8.88%)
    Degree 2^3: 3363682 (14.79%)
    Degree 2^4: 5077860 (22.33%)
    Degree 2^5: 3933648 (17.30%)
    Degree 2^6: 1471050 (6.47%)
    Degree 2^7: 274625 (1.21%)
    Degree 2^8: 111677 (0.49%)
    Degree 2^9: 57589 (0.25%)
    Degree 2^10: 13597 (0.06%)
    Degree 2^11: 7749 (0.03%)
    Degree 2^12: 22 (0.00%)
    Degree 2^13: 3 (0.00%)

  Converting 22744080 vertices, 631153669 directed edges (unordered tuples) to CSR format...
Done converting (21s).
Using random source vertex for each run
Source vertex: 0
Using 2 GPU(s): [ 0 1 ].
Partition begin. seed=1475611716
partition end. (4814.739943 ms)
src_seed = 1475611723
Using traversal-mode LB
__________________________
--------------------------
iteration 0 elapsed: 0.952959 ms, src = 5126276, #iteration = 6
__________________________
[../../gunrock/app/enactor_loop.cuh, 698 @ gpu 0] cudaStreamSynchronize failed (CUDA error 77: an illegal memory access was encountered)
(CUDA error 77 @ GPU 0: an illegal memory access was encountered
(CUDA error 77 @ GPU 0: an illegal memory access was encountered
(CUDA error 77 @ GPU 0: an illegal memory access was encountered
[test_bfs.cu, 478 @ gpu 1] BFS Enact failed (CUDA error 77: an illegal memory access was encountered)

 [BFS] finished.
 avg. elapsed: 0.0000 ms
 iterations: 0
 load time: 26057.3981 ms
 preprocess time: 6269.5360 ms
 postprocess time: 0.0000 ms
 total time: 32437.3341 ms
```

**Comment by sgpyc** [comment]
Should be solved by 23490d3, still don't know why vid_hashtable got out-of-bound accesses.
Can close this issue when no more illegal memory accesses occur for BFS.

updated run result:
```
./bin/test_bfs_8.0_x86_64 market /data/gunrock_dataset/large/arabic-2005/arabic-2005.mtx --src=randomize2 --queue-sizing=6.5 --in-sizing=4 --iteration-num=16 --idempotence --64bit-SizeT --device=0,1 --do_a=100 --do_b=100 --partition-seed=1475611716 --src-seed=1475611723
Loading Matrix-market coordinate-formatted graph ...
  Reading directly from stored binary CSR arrays ...
Done reading (0s).

Degree Histogram (22744080 vertices, 631153669 edges):
    Degree   0: 3425212 (15.06%)
    Degree 2^0: 1733525 (7.62%)
    Degree 2^1: 1253448 (5.51%)
    Degree 2^2: 2020393 (8.88%)
    Degree 2^3: 3363682 (14.79%)
    Degree 2^4: 5077860 (22.33%)
    Degree 2^5: 3933648 (17.30%)
    Degree 2^6: 1471050 (6.47%)
    Degree 2^7: 274625 (1.21%)
    Degree 2^8: 111677 (0.49%)
    Degree 2^9: 57589 (0.25%)
    Degree 2^10: 13597 (0.06%)
    Degree 2^11: 7749 (0.03%)
    Degree 2^12: 22 (0.00%)
    Degree 2^13: 3 (0.00%)

  Converting 22744080 vertices, 631153669 directed edges (unordered tuples) to CSR format...
Done converting (22s).
Using random source vertex for each run
Source vertex: 0
Using 2 GPU(s): [ 0 1 ].
Partition begin. seed=1475611716
partition end. (5229.220152 ms)
src_seed = 1475611723
Using traversal-mode LB
__________________________
--------------------------
iteration 0 elapsed: 1.015186 ms, src = 5126276, #iteration = 6
__________________________
--------------------------
iteration 1 elapsed: 97.419024 ms, src = 22411280, #iteration = 52
__________________________
--------------------------
iteration 2 elapsed: 98.359823 ms, src = 5565530, #iteration = 56
__________________________
--------------------------
iteration 3 elapsed: 3.429890 ms, src = 6734820, #iteration = 18
__________________________
--------------------------
iteration 4 elapsed: 98.104000 ms, src = 1320918, #iteration = 57
__________________________
--------------------------
iteration 5 elapsed: 97.887993 ms, src = 12218967, #iteration = 56
__________________________
--------------------------
iteration 6 elapsed: 98.528147 ms, src = 7611035, #iteration = 58
__________________________
--------------------------
iteration 7 elapsed: 98.699808 ms, src = 14714318, #iteration = 57
__________________________
--------------------------
iteration 8 elapsed: 97.331047 ms, src = 20671983, #iteration = 52
__________________________
--------------------------
iteration 9 elapsed: 99.623919 ms, src = 12843371, #iteration = 62
__________________________
--------------------------
iteration 10 elapsed: 3.022909 ms, src = 14472067, #iteration = 17
__________________________
--------------------------
iteration 11 elapsed: 98.520994 ms, src = 16672038, #iteration = 56
__________________________
--------------------------
iteration 12 elapsed: 1.009941 ms, src = 7219129, #iteration = 6
__________________________
--------------------------
iteration 13 elapsed: 99.057913 ms, src = 1903051, #iteration = 58
__________________________
--------------------------
iteration 14 elapsed: 97.718954 ms, src = 2222582, #iteration = 53
__________________________
--------------------------
iteration 15 elapsed: 97.789049 ms, src = 19679186, #iteration = 55
Computing reference value ...
CPU BFS finished in 1819.568115 msec. cpu_search_depth: 55

Label Validity: 
CORRECT

First 40 labels of the GPU result:
[0:9 1:10 2:13 3:13 4:12 5:13 6:13 7:13 8:12 9:13 10:13 11:12 12:13 13:13 14:13 15:12 16:13 17:13 18:12 19:12 20:13 21:12 22:13 23:12 24:11 25:12 26:12 27:12 28:12 29:12 30:10 31:12 32:11 33:11 34:11 35:11 36:12 37:12 38:12 39:12 ]

	Memory Usage(B)	 #keys0,0	 #keys0,1 #keys1,0	 #keys1,1	 #ins1,0	 #ins1,1 #keys2
GPU_0	 5413142528	 73918262	 73918262	 53869129	 53869129	 33150232	 33150232	 127787389
GPU_1	 5413011456	 73918262	 73918262	 53859814	 53859814	 33144500	 33144500	 127778074
	 queue_sizing =	 6.500000 	 6.500000	 in_sizing =	 4.000000

 [BFS] finished.
 avg. elapsed: 74.2199 ms
 iterations: 56
 min. elapsed: 1.0099 ms
 max. elapsed: 99.6239 ms
 rate: 8325.1943 MiEdges/s
 src: 0
 nodes_visited: 22359925
 edges_visited: 617895191
 nodes queued: 22682124
 edges queued: 629150670
 nodes redundance: 1.44%
 edges redundance: 1.82%
 load time: 26283.7808 ms
 preprocess time: 7963.9962 ms
 postprocess time: 230.5369 ms
 total time: 37534.6711 ms
```

**Comment by yzhwang** [comment]
confirmed that illegal memory access error is gone.
